spec/models/certificate_in_use_event_spec.rb

```diff
@@ -1,12 +1,231 @@
+    certificate = create(:upload_certificate_event, component: old_signing_certificate.component).certificate
+    component = certificate.component
+
+    expected_body = {
+      email_address: 'test@test.com',
+      template_id: '8342fbc4-a847-4587-932c-07065d471942',
+      personalisation: {
+        team_name: component.team.name,
+        component: component.display_long_name,
+        environment: component.environment,
+        time_and_date: component.enabled_signing_certificates.second&.x509&.not_after,
+      }
+    }
+    expect(expected_body[:personalisation][:time_and_date]).to be_present
+    expect(stub_notify_request(expected_body)).to have_been_made.once
+    expect(certificate.events.map(&:class)).to eq [UploadCertificateEvent, CertificateInUseEvent, CertificateNotificationSentEvent]
+  end
+  it 'emails notification with a template without defined deadline when msa signing certificate is uploaded' do
+    stub_cognito_response(method: :list_users_in_group, payload: cognito_users)
+    stub_notify_response
+
+    expect(subject.class).to receive(:create).with(any_args).and_call_original.at_least(:once)
+    certificate = create(:upload_certificate_event, component: create(:msa_component)).certificate
+    component = certificate.component
+
+    expected_body = {
+      email_address: 'test@test.com',
+      template_id: 'ib86fd33c-59c1-4ea4-b643-4a88756c21eb',
+      personalisation: {
+        team_name: component.team.name,
+        component: component.display_long_name,
+        environment: component.environment,
+        time_and_date: component.enabled_signing_certificates.second&.x509&.not_after,
+      }
+    }
+    expect(expected_body[:personalisation][:time_and_date]).to be_nil
+    expect(stub_notify_request(expected_body)).to have_been_made.once
+    expect(certificate.events.map(&:class)).to eq [UploadCertificateEvent, CertificateInUseEvent, CertificateNotificationSentEvent]
+  end
+  it 'emails notification with a template without defined deadline when sp signing certificate is uploaded' do
+    stub_cognito_response(method: :list_users_in_group, payload: cognito_users)
+    stub_notify_response
+
+    expect(subject.class).to receive(:create).with(any_args).and_call_original.at_least(:once)
+    sp_component = create(:sp_component)
+    create(:assign_sp_component_to_service_event, sp_component_id: sp_component.id)
+    certificate = create(:upload_certificate_event, component: sp_component).certificate
+    component = certificate.component
+
+    expected_body = {
+      email_address: 'test@test.com',
+      template_id: 'a07ac619-de15-4bde-97cd-7c722f2b950b',
+      personalisation: {
+        team_name: component.team.name,
+        component: component.display_long_name,
+        environment: component.environment,
+        time_and_date: component.enabled_signing_certificates.second&.x509&.not_after,
```

Comment:
```suggestion
        time_and_date: component.enabled_signing_certificates.second.x509.not_after,
```
this is a test and we should know the values and the variable exist, so no need for `&`'s

Reply:
Thanks, I just copied and pasted. Nice spot